### PR TITLE
Add support for Acer Chromebook 515 Plus (CB515-2H) / Google Omnigul Platform

### PR DESCRIPTION
**Pull request by mdrobnak**
Implementation of Issue https://github.com/linuxboot/heads/issues/1663.

This is implemented using the @MrChromebox Coreboot. Linux 6.1.90, Linux 6.5.13, 6.6.30 all work to recognize the UFS storage. This PR uses 6.6.30.

Version 4.22.01 of Coreboot does not have the Intel ME, VBT, or Flash Descriptor, thus MrChromeBox is used.

Items tested:
- [x] GOP Driver working. No issues booting to Qubes or Ubuntu, splash screen works.
- [x] Update checksums and sign all files in /boot
- [x] TPM Measured Boot
- [x] TOTP Code
- [ ] HOTP Code (No key to test with)
- [ ] TPM Disk Encryption Key does **not** work.
- [x] Flash Upgrade from UI with Config Save
- [x] USB Boot
- [x] System Info - Partial. Disk Size incorrect.
- [x] Generate new TOTP/HOTP Secret
- [x] Reset TPM
- [x] Change Configuration Settings [ Not all Options Tested ]
- [x] Enable Debug
- [x] Add / Remove GPG Key from BIOS
- [x] Power Down

Issues:
* Sealing Disk Unlock Key into TPM does not work.

Configuration Sources:
Linux Configuration based on linux-nitropad-x.config
Coreboot Configuration based on MrChromeBox UEFI configuration. Modified to have Linux payload as per the wiki's instructions.

### CODE REVIEW

**Comment by tlaurion** [comment]
@mdrobnak nice!

- sign your commits and do `commit --signoff` 
- add the board under .circleci/config.yml

We will move from there? You can also join matrix channel and ping me from there, while I prefer PR to be examples for boards porting.

File: boards/omnigul/omnigul.config [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+export CONFIG_BOOT_DEV="/dev/sda2"
+export CONFIG_BOOT_KERNEL_ADD=""
+export CONFIG_BOOT_KERNEL_REMOVE=""
+export CONFIG_BOOTSCRIPT=/bin/gui-init
+
+# Boot Security / Flash Options
+export CONFIG_BOOT_REQ_HASH=n
+export CONFIG_BOOT_REQ_ROLLBACK=n
+export CONFIG_BOARD_NAME="Google Omnigul"
+export CONFIG_FLASHROM_OPTIONS="--noverify-all -p internal --ifd --image bios"
```
Comment (tlaurion) [inline boards/omnigul/omnigul.config:44]: ideally since me is packed, ifd is packed, we would not go "legacy" boards here and prepare images with all blbs required to flash the whole spi, not just the bios region. 

I haven't checked the specs of the baord yet, but iGPU.dGPU and firmwares their firmware will probably need to be added, otherwise jumping from Heads to final OS in kexec call will be problematic to say the least. 

Can you post specs on the PR as well, or add them in the board config?

File: modules/linux [hunk 2]
```diff
@@ -34,6 +34,9 @@ linux_hash := 3991a9e16a187d78d5f414d89236ae5d7f404a69e60c4c42a9d262ee19612ef4
 else ifeq "$(CONFIG_LINUX_VERSION)" "6.1.8"
 linux_version := 6.1.8
 linux_hash := b60bb53ab8ba370a270454b11e93d41af29126fc72bd6ede517673e2e57b816d
+else ifeq "$(CONFIG_LINUX_VERSION)" "6.5.13"
+linux_version := 6.5.13
+linux_hash := 78fbd43822f4c56bc16e89e8874767f592532e1a0ffcd1af4dd279559b5fcbb5
```
Comment (tlaurion) [inline modules/linux:39]: Some patches under patches/linx-version will be needed that's for sure.

File: boards/omnigul/omnigul.config [hunk 3]
```diff
@@ -0,0 +1,52 @@
+# Configuration for a Acer Chromebook Plus 515 (CB515-2H/OMNIGUL)
```
Comment (tlaurion) [inline boards/omnigul/omnigul.config:1]: Document HCL of latop including graphic card and specs for your owned model.

**Review by tlaurion — CHANGES_REQUESTED** [review]
Thanks for this contribution! now let's make this build! Put as draft this PR until ready to be tested by others and to prevent people from bricking their laptop until ready!

File: boards/omnigul/omnigul.config (hunk 1 above)
Comment (mdrobnak) [reply boards/omnigul/omnigul.config:44]: Any guidance on extracting the other firmware would be good. I have a stock Google ROM to work with that I took when I first flashed it to Coreboot before moving it to heads.

Not sure what you mean about the issues with kexec - this setup as posted works for both Ubuntu 23.10 and Qubes 4.2.

Let me know what specs you want, including lspci, dmidecode etc...

**Comment by mdrobnak** [comment]
> @mdrobnak nice!
> 
> * sign your commits and do `commit --signoff`
> * add the board under .circleci/config.yml
> 
> We will move from there? You can also join matrix channel and ping me from there, while I prefer PR to be examples for boards porting.

Ok I'll get this going shortly.

File: boards/omnigul/omnigul.config (hunk 3 above)
Comment (mdrobnak) [reply boards/omnigul/omnigul.config:1]: I attached it. Let me know what else is needed.

File: modules/linux (hunk 2 above)
Comment (mdrobnak) [reply modules/linux:39]: Ah, I see the patches for 6.1.8. I initially picked 6.5 as that was what Ubuntu 23.10 was using - any recommendation on a particular version to use? How was 6.1.8 chosen?

File: modules/linux [hunk 4]
```diff
@@ -34,6 +34,12 @@ linux_hash := 3991a9e16a187d78d5f414d89236ae5d7f404a69e60c4c42a9d262ee19612ef4
 else ifeq "$(CONFIG_LINUX_VERSION)" "6.1.8"
 linux_version := 6.1.8
 linux_hash := b60bb53ab8ba370a270454b11e93d41af29126fc72bd6ede517673e2e57b816d
+else ifeq "$(CONFIG_LINUX_VERSION)" "6.1.90"
```
Comment (tlaurion) [inline modules/linux:37]: Remove if unneeded

File: config/coreboot-omnigul.config [hunk 5]
```diff
@@ -0,0 +1,1473 @@
...
+#
+# Verified Boot (vboot)
+#
+CONFIG_VBOOT_LIB=y
+# end of Verified Boot (vboot)
+
+CONFIG_TPM_GOOGLE=y
+CONFIG_TPM_GOOGLE_CR50=y
+CONFIG_TPM_GOOGLE_IMMEDIATELY_COMMIT_FW_SECDATA=y
+CONFIG_GOOGLE_TPM_IRQ_TIMEOUT_MS=100
```
Comment (tlaurion) [inline config/coreboot-omnigul.config:1290]: Might be too small. Is there timeouts in dmesg?

**Comment by mdrobnak** [comment]
Disk Unlock:
```
user@heads-devel:/media$ cat cbmem_L.log 
TPM2 log:
	Specification: 2.00
	Platform class: PC Client
TPM2 log entry 1:
	PCR: 2
	Event type: Action
	Digests:
		 SHA256: 648fd22dbf689a7facd6f6935170dd740bce45c36a7831032e71eb180078ab0c
	Event data: FMAP: FMAP
TPM2 log entry 2:
	PCR: 2
	Event type: Action
	Digests:
		 SHA256: 03796d2e280f07673cb9f6c013443378ffb8290ea709babb83550af82fc940cd
	Event data: CBFS: bootblock
TPM2 log entry 3:
	PCR: 2
	Event type: Action
	Digests:
		 SHA256: e2fee1a8472d4986e5a142c5d15f162835854ea772f0e86d1379a554d4f4ea21
	Event data: CBFS: fallback/romstage
TPM2 log entry 4:
	PCR: 2
	Event type: Action
	Digests:
		 SHA256: 20f555dbb6678b91b75f4d635df597bca7cc6cf2ff477e22bc6e1df6bbea61cb
	Event data: CBFS: fspm.bin
TPM2 log entry 5:
	PCR: 2
	Event type: Action
	Digests:
		 SHA256: 7c52722fdf3eb612144c9e3747fe05dbbcf9e02811a505f6cdd33494da78fc1d
	Event data: CBFS: spd.bin
TPM2 log entry 6:
	PCR: 2
	Event type: Action
	Digests:
		 SHA256: 388468b7b064bf27b13e49d91f6c80d6313a1ea226d9ebfc82aa260382d1bb30
	Event data: CBFS: fallback/postcar
TPM2 log entry 7:
	PCR: 2
	Event type: Action
	Digests:
		 SHA256: ad2053fd5b97f52d3a25edea9500ba0f697accd467d9a6e0f7a3bc22c0a8270b
	Event data: CBFS: fallback/ramstage
TPM2 log entry 8:
	PCR: 2
	Event type: Action
	Digests:
		 SHA256: 970fbc1d69d801604a09967036dbf1c07c514c30a2835a4a53a5295372346b88
	Event data: CBFS: cpu_microcode_blob.bin
TPM2 log entry 9:
	PCR: 2
	Event type: Action
	Digests:
		 SHA256: 3dc0ba24e5dd1c8a2d7880cf6e40e169eb92e193b68d838ad69be08f6cc9a2a6
	Event data: CBFS: fsps.bin
TPM2 log entry 10:
	PCR: 2
	Event type: Action
	Digests:
		 SHA256: 83c94bebba28305531ac7fa841f5075754fec72582fb8ef770e81bd430aadc49
	Event data: CBFS: vbt.bin
TPM2 log entry 11:
	PCR: 2
	Event type: Action
	Digests:
		 SHA256: c9b940f1866ecd4fb8f23a622a17b1b42b59be2dff09d7bc8e8e1a9fb62a8c5b
	Event data: CBFS: fallback/dsdt.aml
TPM2 log entry 12:
	PCR: 2
	Event type: Action
	Digests:
		 SHA256: 2f045194201a13dcffaadf1b295b5d8b156a51e7ca8fb93b05c8927bc2391d9e
	Event data: CBFS: bootsplash.jpg
TPM2 log entry 13:
	PCR: 2
	Event type: Action
	Digests:
		 SHA256: 42e8453008e55c6383763e0ea91699203443bc52ccb259e8185a3ae3d861d3f8
	Event data: CBFS: fallback/payload
user@heads-devel:/media$ 
```

Attaching debug.log as that's larger.
[debug.log](https://github.com/linuxboot/heads/files/15473300/debug.log)

File: modules/linux (hunk 4 above)
Comment (mdrobnak) [reply modules/linux:37]: Missed that during my cleanup. Will do.

File: config/coreboot-omnigul.config (hunk 5 above)
Comment (mdrobnak) [reply config/coreboot-omnigul.config:1290]: Not that I've seen. I can try modifying that.

**Comment by tlaurion** [comment]
As written under matrix thread for this updated PR at  https://matrix.to/#/!pAlHOfxQNPXOgFGTmo:matrix.org/$VKP1ynETRMavsOqy3SqUf_nadPUgyoSC9H_13y8Dpjo?via=matrix.org&via=nitro.chat&via=fairydust.space


The white rabbit to be followed is why CR50 TPM refuses to  to add TPM DUK nv region into TPM which doesn't seem supported on CR50 not sure why:

```
TRACE: /bin/tpmr(32): main
TRACE: /bin/tpmr(413): tpm2_seal
DEBUG: tpm2_seal: file=/tmp/secret/secret.key handle=0x81000003 pcrl=0,1,2,3,4,5,6,7 pcrf=/tmp/secret/pcrf.bin pass=<hidden>
LOG: tpmr stderr: WARNING:esys:src/tss2-esys/api/Esys_PolicyPassword.c:292:Esys_PolicyPassword_Finish() Received TPM Error 
LOG: tpmr stderr: ERROR:esys:src/tss2-esys/api/Esys_PolicyPassword.c:106:Esys_PolicyPassword() Esys Finish ErrorCode (0x000b0143) 
LOG: tpmr stderr: ERROR: Esys_PolicyPassword(0xB0143) - rmt:error(2.0): command code not supported
LOG: tpmr stderr: ERROR: Could not build policyauthvalue TPM
LOG: tpmr stderr: ERROR: Unable to run policypassword
```

File: boards/omnigul/omnigul.config [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+export CONFIG_COREBOOT_VERSION=mrchromebox
+export CONFIG_LINUX_VERSION=6.6.30
+
+CONFIG_COREBOOT_CONFIG=config/coreboot-omnigul.config
+CONFIG_LINUX_CONFIG=config/linux-omnigul.config
+
+#Enable DEBUG output
+#export CONFIG_DEBUG_OUTPUT=y
+#export CONFIG_ENABLE_FUNCTION_TRACING_OUTPUT=y
+#Enable TPM2 pcap output under /tmp
```
Comment (tlaurion) [inline boards/omnigul/omnigul.config:16]: @mdrobnak until this port is successful, I think you might want to add a commit turning this on and a final one turning this off

**Comment by mdrobnak** [comment]
Attached are two files:
[debug_reset_tpm_sign_reboot.log](https://github.com/linuxboot/heads/files/15475804/debug_reset_tpm_sign_reboot.log)
That file is boot, reset TPM, sign /boot files, copy logs, reboot.

[debug_after_reboot_attempt_duk_seal.log](https://github.com/linuxboot/heads/files/15475812/debug_after_reboot_attempt_duk_seal.log)
After rebooting, go to OS boot menu, set default, attempt to add Disk Unlock Key.

**Comment by tlaurion** [comment]
TLDR: Chromebooks have a fTPM which differs from dTPM (documentation and understanding is missing to be supported correctly from the TPM toolstack), correct @mdrobnak ?


Putting this as draft until this is worked seperately.

**Comment by tlaurion** [comment]
@mdrobnak revisiting because interest at https://matrix.to/#/!pAlHOfxQNPXOgFGTmo:matrix.org/$E4N2td-clfQq3IP4F4v_04oSLABpj-Kcm5rNVvDgi30?via=matrix.org&via=nitro.chat&via=envs.net

Not all boards support TPM DUK, and can today be deactivated as a possibility through `CONFIG_TPM_NO_LUKS_DISK_UNLOCK` that can be added into board config as can be seen checked at https://github.com/linuxboot/heads/blob/b28c257b851c880f5dc2c4adf0c47377026529cb/initrd/bin/kexec-save-default#L190


Until TPM DUK (seperate secret sealing nvram region issue figured out), Chromebooks **could** be supported with this bfeature missing? Thoughts?

**Comment by mdrobnak** [comment]
> @mdrobnak revisiting because interest at https://matrix.to/#/!pAlHOfxQNPXOgFGTmo:matrix.org/$E4N2td-clfQq3IP4F4v_04oSLABpj-Kcm5rNVvDgi30?via=matrix.org&via=nitro.chat&via=envs.net
> 
> Not all boards support TPM DUK, and can today be deactivated as a possibility through `CONFIG_TPM_NO_LUKS_DISK_UNLOCK` that can be added into board config as can be seen checked at
> 
> https://github.com/linuxboot/heads/blob/b28c257b851c880f5dc2c4adf0c47377026529cb/initrd/bin/kexec-save-default#L190
> 
> Until TPM DUK (seperate secret sealing nvram region issue figured out), Chromebooks **could** be supported with this bfeature missing? Thoughts?

I think that makes perfect sense to do this.

I am updating my PR as I just rebased it off master a few minutes ago, and it seems fine at first glance.

-Matt

**Comment by tlaurion** [comment]
@mdrobnak sorry for the delay. If you could rebase this on master, I think that could be reviewed and merged!

**Comment by tlaurion** [comment]
@mdrobnak ping

**Comment by mdrobnak** [comment]
@tlaurion  Sorry, meant to respond earlier. It's been a bit hectic here - I have it on my To-Do list for this week.

**Comment by mdrobnak** [comment]
I rebased the bracnh, haven't tested a new flash yet. Should be able to do that tomorrow at some point.

**Comment by tlaurion** [comment]
Pushed changes in branch https://github.com/tlaurion/heads/tree/mdrobnak-google_omnigul

revert linux to 6.1.8, bump mrchromebox to 24.02, add patches of 24.02.01, uniformize board config, add PR0 requirements (to be tested)

@mdrobnak cherry-pick, amend merge after testing.

**Comment by tlaurion** [comment]
> Initial test build met with error, I haven't gotten back to it yet.
> 
> ```
> 2025-03-21 15:01:26+00:00 CPIO      build/x86/omnigul/tools.cpio
> 2025-03-21 15:01:26+00:00 UNCHANGED build/x86/omnigul/tools.cpio
> e66f30bcd375d1d5f16e919513480b4fe61dbfd68a6e0d84dfa4c3ef308aa83f  /ssdpool/home/mdrobnak/heads-google/build/x86/omnigul/tools.cpio
> 18902528:/ssdpool/home/mdrobnak/heads-google/build/x86/omnigul/tools.cpio
> 2025-03-21 15:01:27+00:00 HASHES    build/x86/omnigul/tools.cpio
> 2025-03-21 15:01:27+00:00 SIZES     build/x86/omnigul/tools.cpio
> 2025-03-21 15:01:27+00:00 CPIO      build/x86/omnigul/heads.cpio
> 2025-03-21 15:01:27+00:00 UNCHANGED build/x86/omnigul/heads.cpio
> 1672750cf86abf26d685f67dc8a19d6a08f11868ae42ac17b06ebbb684f86c89  /ssdpool/home/mdrobnak/heads-google/build/x86/omnigul/heads.cpio
>   426496:/ssdpool/home/mdrobnak/heads-google/build/x86/omnigul/heads.cpio
> 2025-03-21 15:01:27+00:00 HASHES    build/x86/omnigul/heads.cpio
> 2025-03-21 15:01:27+00:00 SIZES     build/x86/omnigul/heads.cpio
> 2025-03-21 15:01:27+00:00 MAKE coreboot-mrchromebox
> tail /ssdpool/home/mdrobnak/heads-google/build/x86/log/coreboot-mrchromebox.log
> -----
> cc1: all warnings being treated as errors
> cc1: all warnings being treated as errors
> firmware/2lib/2misc.c: In function 'vb2_validate_gbb_signature':
> firmware/2lib/2misc.c:23:9: error: 'static' is not at beginning of declaration [-Werror=old-style-declaration]
>    23 |         const static uint8_t sig_xor[VB2_GBB_SIGNATURE_SIZE] =
>       |         ^~~~~
> make[2]: *** [Makefile:1163: /ssdpool/home/mdrobnak/heads-google/build/x86/coreboot-mrchromebox/omnigul/external/vboot_reference-postcar/firmware/lib/cgptlib/cgptlib_internal.o] Error 1
> make[2]: *** Waiting for unfinished jobs....
> cc1: all warnings being treated as errors
> make[2]: *** [Makefile:1164: /ssdpool/home/mdrobnak/heads-google/build/x86/coreboot-mrchromebox/omnigul/external/vboot_reference-romstage/firmware/lib/cgptlib/cgptlib_internal.o] Error 1
> make[2]: *** Waiting for unfinished jobs....
> make[2]: *** [Makefile:1163: /ssdpool/home/mdrobnak/heads-google/build/x86/coreboot-mrchromebox/omnigul/external/vboot_reference-ramstage/firmware/2lib/2misc.o] Error 1
> make[1]: *** [src/security/vboot/Makefile.mk:54: omnigul/external/vboot_reference-ramstage/vboot_fw.a] Error 2
> cc1: all warnings being treated as errors
> make[2]: *** [Makefile:1163: /ssdpool/home/mdrobnak/heads-google/build/x86/coreboot-mrchromebox/omnigul/external/vboot_reference-postcar/firmware/2lib/2misc.o] Error 1
> make[1]: *** [src/security/vboot/Makefile.mk:55: omnigul/external/vboot_reference-postcar/vboot_fw.a] Error 2
> cc1: all warnings being treated as errors
> make[2]: *** [Makefile:1163: /ssdpool/home/mdrobnak/heads-google/build/x86/coreboot-mrchromebox/omnigul/external/vboot_reference-romstage/firmware/2lib/2misc.o] Error 1
> make[1]: *** [src/security/vboot/Makefile.mk:52: omnigul/external/vboot_reference-romstage/vboot_fw.a] Error 2
> make[1]: Leaving directory '/ssdpool/home/mdrobnak/heads-google/build/x86/coreboot-mrchromebox'
> make: *** [Makefile:570: /ssdpool/home/mdrobnak/heads-google/build/x86/coreboot-mrchromebox/omnigul/.build] Error 1
> ```
> 
> -Matt

Sorry about bitrotting things once more. T480 and all other board bumped to coreboot 24.12, so I did so and left commit log notes for you to review under https://github.com/tlaurion/heads/commit/c03aeafb72a9014b09d5f20cbaac52f07b5b869c which should build (it built locally). 

So if we act somehow fast here (not another 6 months) maintaining this board once merged should be easy enough.

Please see testing notes under pending for merge doc per https://github.com/linuxboot/heads-wiki/pull/184
- rendered at https://notgivenby.github.io/heads-wiki/Porting/ until merged
  - Specifically for https://notgivenby.github.io/heads-wiki/Porting/#testing

Extract here for testing todo:
- [ ] Successful external flash link to circleci: https://X.Y.Z. from commit `X.Y.Z.` using external programmer model `X.Y.Z.` on `X.Y.Z.` Voltage mode
- [ ] Boots successfully after the flashing: 
- [ ] Setting clock prompt on first reboot: ok if triggered correctly after initial flashing and CMOS battery disconnected
- [ ] Clean boot detected (no keyring, nothing installed on disk): usb boot proposed and followed
- [ ] Boots on usb
- [ ] OS `X.Y.Z.` install and reboot
- [ ] Heads functionality- no pubkey detected, but OS detected -> OEM-Factory-reset proposed. Done with `X.Y.Z.` hardwarekey e.g. nk3
- [ ] On reboot after re-ownership: generate new HOTP/TOTP
- [ ] ~On reboot: default boot proposes to choose boot default + TPM DUK~ : No TPM DUK for CR50 based TPMs for now
- [ ] ~TPM DUK boots OS version `X.Y.Z.`~ : No TPM DUK for CR50 based TPMs for now
- [ ] ~TPM DUK boots QubesOS `X.Y.Z.`~ : No TPM DUK for CR50 based TPMs for now
- [ ] wifi works based on OS `X.Y.Z.`
- [ ] PR0
  - [ ] flashprog -p internal (not locked)
  - [ ] lock_chip (locks the platform with PR0, if PR0 patch applied in fork or under `patches/coreboot-X.Y.Z.` and coreboot config contain proper preparation of the platform)
  - [ ] flashprog -p internal (reports locked)

CC @mdrobnak

**Comment by tlaurion** [comment]
> Initial test build met with error, I haven't gotten back to it yet.
> 
> ```
> 2025-03-21 15:01:26+00:00 CPIO      build/x86/omnigul/tools.cpio
> 2025-03-21 15:01:26+00:00 UNCHANGED build/x86/omnigul/tools.cpio
> e66f30bcd375d1d5f16e919513480b4fe61dbfd68a6e0d84dfa4c3ef308aa83f  /ssdpool/home/mdrobnak/heads-google/build/x86/omnigul/tools.cpio
> 18902528:/ssdpool/home/mdrobnak/heads-google/build/x86/omnigul/tools.cpio
> 2025-03-21 15:01:27+00:00 HASHES    build/x86/omnigul/tools.cpio
> 2025-03-21 15:01:27+00:00 SIZES     build/x86/omnigul/tools.cpio
> 2025-03-21 15:01:27+00:00 CPIO      build/x86/omnigul/heads.cpio
> 2025-03-21 15:01:27+00:00 UNCHANGED build/x86/omnigul/heads.cpio
> 1672750cf86abf26d685f67dc8a19d6a08f11868ae42ac17b06ebbb684f86c89  /ssdpool/home/mdrobnak/heads-google/build/x86/omnigul/heads.cpio
>   426496:/ssdpool/home/mdrobnak/heads-google/build/x86/omnigul/heads.cpio
> 2025-03-21 15:01:27+00:00 HASHES    build/x86/omnigul/heads.cpio
> 2025-03-21 15:01:27+00:00 SIZES     build/x86/omnigul/heads.cpio
> 2025-03-21 15:01:27+00:00 MAKE coreboot-mrchromebox
> tail /ssdpool/home/mdrobnak/heads-google/build/x86/log/coreboot-mrchromebox.log
> -----
> cc1: all warnings being treated as errors
> cc1: all warnings being treated as errors
> firmware/2lib/2misc.c: In function 'vb2_validate_gbb_signature':
> firmware/2lib/2misc.c:23:9: error: 'static' is not at beginning of declaration [-Werror=old-style-declaration]
>    23 |         const static uint8_t sig_xor[VB2_GBB_SIGNATURE_SIZE] =
>       |         ^~~~~
> make[2]: *** [Makefile:1163: /ssdpool/home/mdrobnak/heads-google/build/x86/coreboot-mrchromebox/omnigul/external/vboot_reference-postcar/firmware/lib/cgptlib/cgptlib_internal.o] Error 1
> make[2]: *** Waiting for unfinished jobs....
> cc1: all warnings being treated as errors
> make[2]: *** [Makefile:1164: /ssdpool/home/mdrobnak/heads-google/build/x86/coreboot-mrchromebox/omnigul/external/vboot_reference-romstage/firmware/lib/cgptlib/cgptlib_internal.o] Error 1
> make[2]: *** Waiting for unfinished jobs....
> make[2]: *** [Makefile:1163: /ssdpool/home/mdrobnak/heads-google/build/x86/coreboot-mrchromebox/omnigul/external/vboot_reference-ramstage/firmware/2lib/2misc.o] Error 1
> make[1]: *** [src/security/vboot/Makefile.mk:54: omnigul/external/vboot_reference-ramstage/vboot_fw.a] Error 2
> cc1: all warnings being treated as errors
> make[2]: *** [Makefile:1163: /ssdpool/home/mdrobnak/heads-google/build/x86/coreboot-mrchromebox/omnigul/external/vboot_reference-postcar/firmware/2lib/2misc.o] Error 1
> make[1]: *** [src/security/vboot/Makefile.mk:55: omnigul/external/vboot_reference-postcar/vboot_fw.a] Error 2
> cc1: all warnings being treated as errors
> make[2]: *** [Makefile:1163: /ssdpool/home/mdrobnak/heads-google/build/x86/coreboot-mrchromebox/omnigul/external/vboot_reference-romstage/firmware/2lib/2misc.o] Error 1
> make[1]: *** [src/security/vboot/Makefile.mk:52: omnigul/external/vboot_reference-romstage/vboot_fw.a] Error 2
> make[1]: Leaving directory '/ssdpool/home/mdrobnak/heads-google/build/x86/coreboot-mrchromebox'
> make: *** [Makefile:570: /ssdpool/home/mdrobnak/heads-google/build/x86/coreboot-mrchromebox/omnigul/.build] Error 1
> ```
> 
> -Matt

(As for reason of why vboot failed: it was because the commit searched by coreboot fork used commit to be used for vboot submodule didn't exist. Cannot report bug for old version, but modules/coreboot 24.12, reusing 24.12 buildstack for all most Heads used boards upstream were bumped to 24.12 under https://github.com/linuxboot/heads/pull/1908 so I invested time moving this forward instead of reporting bug for 24.02.01 which is now more than year old (Coreboot release versioning scheme changed to YY.MM: 24.02 is Feb 2024, 24.12 is December 24.12: most downstream coreboot forks used into Heads are either under 24.12 or will be soon, so it was logical for me to do this instead of fixing what doen't work in downstream forks, patching etc).

Sorry again for PR bitrotting side effects. We shouldbe good to merge soon if all is good on your side testing wise @mdrobnak

**Comment by mdrobnak** [comment]
Good news is that initial testing with your updated items does seem to work just fine. I'm hoping to get things cleaned up and run through yout test checklist next week.

**Comment by mdrobnak** [comment]
These are the only options when going through Coreboot that caught my eye, not sure if they are as desired:
```
 [ ] Validate Intel firmware descriptor
 [ ] Lock down chipset in coreboot 
 Protect flash regions (Unlock flash regions)  --->
 [*] Support Intel PCI-e WiFi adapters
 [*] Support MediaTek PCI-e WiFi adapters
 Boot media protection mechanism (Lock boot media using the controller)  --->
 Boot media protected regions (Write-protect the whole boot medium)  --->
 [*] Perform MP Initialization by FSP using coreboot MP PPI service (NEW)
 [*] Total Memory Encryption (TME)/Multi-key TME (MKTME) (NEW)
```

Aside from the version string...that's really the only thing I changed.

As far as that testing guide...I'll re-do the stuff that doesn't require me to takt this thing aapart and re-instrall stuff... I have tested those items in the past and they did work as expected..

-Matt

**Comment by tlaurion** [comment]
> These are the only options when going through Coreboot that caught my eye, not sure if they are as desired:
> 
> ```
>  [ ] Validate Intel firmware descriptor
>  [ ] Lock down chipset in coreboot 
>  Protect flash regions (Unlock flash regions)  --->
>  [*] Support Intel PCI-e WiFi adapters
>  [*] Support MediaTek PCI-e WiFi adapters
>  Boot media protection mechanism (Lock boot media using the controller)  --->
>  Boot media protected regions (Write-protect the whole boot medium)  --->
>  [*] Perform MP Initialization by FSP using coreboot MP PPI service (NEW)
>  [*] Total Memory Encryption (TME)/Multi-key TME (MKTME) (NEW)
> ```
> 
> Aside from the version string...that's really the only thing I changed.
> 
> As far as that testing guide...I'll re-do the stuff that doesn't require me to takt this thing aapart and re-instrall stuff... I have tested those items in the past and they did work as expected..
> 
> -Matt

Will reverify the configs against other boards when I have a chance.

Cannot merge if untested. Cannot test if unsure of configs. Not fresh in my mind anymore, will try to find time to check in the next days.

**Comment by mdrobnak** [comment]
I will leave my branch on my fork up, but at this point I do not have the resources to dedicate to this. It was slightly over a year ago I opened this PR. I appreciate all that has been done on this project, but to me this exercise felt like a constantly changing target. I wish you success with the project. At this time, I'm closing this PR.

**Comment by tlaurion** [comment]
@mdrobnak if I do a superseeding PR with this, is there a date forward you would be willing to test with readiness to revert to backup in chance of a brick or are we willing to leave this work go to waste?

Sorry for the feeling of this being a moving target. I'll try to find time to update the porting guide with timely testing requirements for pull requests to not bitrot otherwise yes, this is a moving target until I can version bump merged stuff all at once across all boards.

If there is no interest in testing this in a timely manner it's OK as well, I'll make sure there is common understanding in the porting guide as well.

**Comment by mdrobnak** [comment]
I tested the initial flash procedures the last time around. I am not reverting the box back to ChromeOS and doing all of this again. I will not re-test that. That is not a small time investment. Sorry.

The main reason for the time delay was the lack of doing disk unlock with the CR50. Had we said "this is OK without this" we would have been done significantly sooner, and we would not be in this spot.

Again, I like the project, I wish you the best, but I'm out.

-Matt
